File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        nDict = {}
        for i in range(len(nums)):
            need = target - nums[i]

            if need in nDict:
                return [nDict[need], i]

            nDict[nums[i]] = i

        return [0, 0]

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int pointer1 = 0;
        int pointer2 = numbers.length -1;

        while(pointer1 < pointer2){
            if(numbers[pointer1] + numbers[pointer2] == target){
                break;
            }
            if(numbers[pointer1] + numbers[pointer2] > target){
                pointer2--;
            }
            if(numbers[pointer1] + numbers[pointer2] < target){
                pointer1++;
            }
        }

        return new int[]{pointer1 + 1, pointer2 + 1};
    }
}
